bindings/python/pyiree/testing/compilation/stages.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Any, Callable, Dict, Optional, Sequence, Tuple
+
+import pyiree as iree
+import pyiree.compiler2
+from ..utils import ParsableConstants
+
+__all__ = [
+    "Representation",
+    "Representations",
+    "Stage",
+    "Stages",
+    "save_split_methods",
+]
+
+
+class Representation:
+
+  def __init__(self, lower_name: str, upper_name: str, extension: str):
+    self.lower_name = lower_name
+    self.upper_name = upper_name
+    self.extension = extension
+    self.file_extension = f"{lower_name}.{extension}"
+    self.methods_extension = f"{lower_name}.methods"
+
+  def __str__(self):
+    return self.lower_name
+
+  def __repr__(self):
+    return (f"Representation({repr(self.lower_name)}, {repr(self.upper_name)}, "
+            f"{repr(self.extension)})")
+
+
+class Representations(ParsableConstants):
+  """Representations for the core IREE compiler."""
+  MHLO = Representation("mhlo", "MHLO", "mlir")
```

Comment:
what uses the upper name? can we remove that? the fewer ways to reference the same thing the better - consistency in tooling would be nice - I'm not sure why the names python stuff uses ever differed but it's been a pain

Reply:
I don't like it either, but the style guide for class names is FirstUpperSnakeCase. We could use the lower name only, if we're okay with class names like `mhlo_to_llvmaot_TestCase` (which is ugly, but _is_ more readable than `MHLOToLLVMAOTTestCase` IMO).

Reply:
You have my official permission to ignore the style guide if it aids pragmatism/consistency/readability :) Style guide always takes a second place to doing the right thing, especially for things meant to be used by humans outside of the project.